cmd/db.go

```diff
@@ -2,29 +2,57 @@ package cmd
 
 import (
 	"fmt"
+	"os"
 
-	"github.com/redhatinsights/ros-ocp-backend/internal/config"
-	"github.com/redhatinsights/ros-ocp-backend/internal/db"
-	"github.com/redhatinsights/ros-ocp-backend/internal/model"
+	"database/sql"
+
+	"github.com/golang-migrate/migrate/v4"
 	"github.com/spf13/cobra"
+
+	"github.com/golang-migrate/migrate/v4/database/pgx"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	"github.com/redhatinsights/ros-ocp-backend/internal/config"
 )
 
-var migrateCmd = &cobra.Command{
-	Use:   "migrate",
-	Short: "migrate database",
+var migrateCmd = &cobra.Command{Use: "migrate", Short: "migrate database"}
+
+var migrateUp = &cobra.Command{
+	Use:   "up",
+	Short: "Forward database migration",
+	Long:  "Forward database migration",
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("Starting database migration")
-		config.InitConfig()
-		db.InitDB()
-		err := db.DB.AutoMigrate(
-			&model.RHAccount{},
-			&model.Cluster{},
-			&model.Workload{},
-			&model.RecommendationSet{},
-		)
+		fmt.Println("Forward database migration")
+		cfg := config.GetConfig()
+		db, err := sql.Open("pgx", fmt.Sprintf("postgres://%s:%s@%s:%s/%s", cfg.DBUser, cfg.DBPassword, cfg.DBHost, cfg.DBPort, cfg.DBName))
+		if err != nil {
+			fmt.Printf("Unable to get *sql.DB: %v\n", err)
+			os.Exit(1)
+		}
+		driver, err := pgx.WithInstance(db, &pgx.Config{})
 		if err != nil {
-			fmt.Println("DB Migration Failed..")
+			fmt.Printf("Unable to get db driver: %v\n", err)
+			os.Exit(1)
 		}
+		m, err := migrate.NewWithDatabaseInstance("file://./migrations", cfg.DBName, driver)
+		if err != nil {
+			fmt.Printf("Unable to get migration instance: %v\n", err)
+			os.Exit(1)
+		}
+		err = m.Up()
+		if err != nil {
+			fmt.Println(err)
+		}
+	},
+}
+
+var migratedown = &cobra.Command{
+	Use:   "down",
+	Short: "Reverse database migration",
+	Long:  "Reverse database migration",
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("Reverse database migration")
+		// Placeholder for db downgrade.
+		// This will be helpful for force unlock dirty migration
 	},
 }
```

Comment:
A couple of command suggestions, I think `head` will be especially useful.

- `revisions` (a list of all applied migrations)
- `head` (details about the latest applied migration)

Reply:
I have added only one sub-command `revision` to get the migration details. 
`golang-migrate` scripts does not add much to `schema_migrations` table to display. 
```
postgres=# select * from schema_migrations ;
 version | dirty 
---------+-------
       4 | f
(1 row)
```

Reply:
Sounds good, as long as we can make a bit of sense about applied/pending migrations :+1:

Reply:
IIUC the `revision` ~ `schema_migrations` should let me know how many migrations are applied atm.
So I went with `down` a few times but the revisions show me 4 everytime, am I missing something?

```
( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db migrate up                                                                             (kind-kind) (ros-ocp-backend) 15:36:11
Forward database migration
no change
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db migrate down                                                                           (kind-kind) (ros-ocp-backend) 15:36:49
Reverse database migration
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db revision                                                                               (kind-kind) (ros-ocp-backend) 15:36:54
Current migration version is: 4 
Is dirty: false 
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db migrate down                                                                           (kind-kind) (ros-ocp-backend) 15:36:58
Reverse database migration
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db revision                                                                               (kind-kind) (ros-ocp-backend) 15:37:02
Current migration version is: 4 
Is dirty: false 
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db migrate down                                                                           (kind-kind) (ros-ocp-backend) 15:37:05
Reverse database migration
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db revision                                                                               (kind-kind) (ros-ocp-backend) 15:37:32
Current migration version is: 4 
Is dirty: false 
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db migrate down                                                                           (kind-kind) (ros-ocp-backend) 15:37:34
Reverse database migration
[I] ( ⌁ ) w ) f ) ros-ocp-backend ) go run rosocp.go db revision                                                                               (kind-kind) (ros-ocp-backend) 15:37:36
Current migration version is: 4 
Is dirty: false 

```

Reply:
I haven't implemented `down` subcommand. I don't want that someone by mistakenly run `down` command in production DB. 
But to test it. you can run `make install-golang-migrate-cli-tool` to install migrate cli tool and then run below command. 

`./bin/migrate -path ./migrations/ -database "postgres://postgres:postgres@localhost:15432/postgres?sslmode=disable" down
`

Reply:
IMO, we need both `up` and `down`, IIUC our PROD DB would be on RDS so the concerned team can manage it.
If not, we can have appropriate documentation later about these.

Reply:
I added the `down` command.